airflow/jobs.py

```diff
@@ -704,7 +704,7 @@ def _execute(self):
         self.logger.info("Starting the scheduler")
 
         dagbag = models.DagBag(self.subdir, sync_to_db=True)
-        executor = self.executor = dagbag.executor
+        executor = self.executor
```

Comment:
Previous behavior was using the `DagBag` executor, but that one is always set to the default executor. (By the way, it looks like the executor of the `DagBag` is never used.)